lib/cocina_checker.rb

```diff
@@ -7,9 +7,13 @@ def self.check
   end
 
   def check_cocina
-    puts "repos to Cocina check: #{repos.map(&:name).join(', ')}"
+    gem_filename = 'Gemfile.lock'
+    repo_names = repos.map(&:name)
+    puts "repos to Cocina check: #{repo_names.join(', ')}"
+    version_map = Dir["#{base_directory}**/#{gem_filename}"].filter_map do |lockfile_path|
+      cached_repo_name = lockfile_path.gsub(base_directory, '').gsub(gem_filename, '').chomp('/')
+      next unless repo_names.include? cached_repo_name # skip any cached repo not found in the settings.yml file
```

Comment:
I would say -- why don't we simply remove the cached repos in the tmp dir with this script instead?  It's a tmp directory in a very tiny app we use to deploy repos.  We're just removing a local copy of the code.